test/fuzz/transformation_wrap_vector_synonym_test.cpp

```diff
@@ -0,0 +1,540 @@
+// Copyright (c) 2021 Shiyu Liu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_wrap_vector_synonym.h"
+#include "gtest/gtest.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "source/fuzz/transformation_composite_construct.h"
+#include "test/fuzz/fuzz_test_util.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+TEST(TransformationWrapVectorSynonym, SimpleTest) {
+  std::string shader = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main" %97
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 310
+               OpName %4 "main"
+               OpName %8 "i1"
+               OpName %10 "i2"
+               OpName %14 "v2"
+               OpName %20 "ui1"
+               OpName %22 "ui2"
+               OpName %26 "v3"
+               OpName %33 "f1"
+               OpName %35 "f2"
+               OpName %39 "v4"
+               OpName %47 "int_add"
+               OpName %51 "int_sub"
+               OpName %55 "int_mul"
+               OpName %59 "int_div"
+               OpName %63 "uint_add"
+               OpName %67 "uint_sub"
+               OpName %71 "uint_mul"
+               OpName %75 "uint_div"
+               OpName %79 "float_add"
+               OpName %83 "float_sub"
+               OpName %87 "float_mul"
+               OpName %91 "float_div"
+               OpName %97 "value"
+               OpDecorate %8 RelaxedPrecision
+               OpDecorate %10 RelaxedPrecision
+               OpDecorate %14 RelaxedPrecision
+               OpDecorate %15 RelaxedPrecision
+               OpDecorate %16 RelaxedPrecision
+               OpDecorate %17 RelaxedPrecision
+               OpDecorate %20 RelaxedPrecision
+               OpDecorate %22 RelaxedPrecision
```

Comment:
Strip out these decorations unless they are relevant to your transformation.

Though this has made me think: if the result id of the scalar operation has the `RelaxedPrecision` decoration, we should probably apply the same decoration to the result id of the corresponding vector operation. What do you think, @Vasniktel? @sliu-UIUC Rather than having this hold up your current PR, you could open an issue for it to sort out in a future PR.

Reply:
That makes sense.